### DISCUSSION TITLE
Typed block port description?

### DISCUSSION THREAD

**Discussion by astrojhgu** [FutureSDR/FutureSDR#79] [Ideas]
Now the port description uses `&str`, e.g., "in", or "in1", "in2", etc. Is there a plan to make it a specific type?

For example
```
enum PortDesc{
  In(usize), 
  Out(usize)
}
```
And the current port description "in", "out" seem to be undocumented.

**Comment by JuliDi** [comment]
Presumably related discussion in #66

**Comment by bastibl** [comment]
Good idea, but I think it should be
```rust
enum PortIdentifier {
  Name(String), 
  Id(usize)
}
```
and with the right `From/Into` implementations, it should still be usable with `"in", "out"` or `0, 1`. And the `connect_stream/connect_message` functions would use `impl Into<PortIdentifier>`. 

Anybody motivated to try this? If you agree that with this idea, we could make it an issue.

**Comment by bastibl** [comment]
The suggestion was implemented here: https://github.com/bastibl/FutureSDR/commit/29659cf136d6a520d9fcb679c70a12a90e66b007. And will be merged upstream shortly.

**Reply by etschneider** [reply]
Totally a topic of a larger discussion, but shouldn't a port ID also identify the block it belongs to? E.g. `connect(src_port_id, dst_port_id)`.
